app/scripts/lib/app-start.js

```diff
@@ -534,21 +534,36 @@ define(function (require, exports, module) {
       return this._user.upgradeFromSession(Session, this._fxaClient);
     },
 
+    viewFactory: function (Constructor, options) {
```

Comment:
I'm not a huge fan of `viewFactory` as a method name. Although that describes what it is, I think it reads less clearly than `createView` when you see it invoked.